### ISSUE TITLE
Custom AuthProvider not working with <HydraAdmin/>

### ISSUE THREAD

**Issue by tmihalicka** [api-platform/admin#551]
**API Platform version(s) affected**: 3.4.6

**Description**  
After upgrading `@api-platform/admin` package to version `3.4.6` from version `3.4.5`, authProvider not working when is used with `<HydraAdmin />`, when i downgrade package to version `3.4.5` everything start works flawlessly

**How to reproduce**  
Add custom `authProvider` to `<HydraAdmin />`

**Additional Context**  
My `<HydraAdmin />` init
```javascript
const App = () => {
  return (
    <Sentry.ErrorBoundary fallback={'An error has occurred'}>
      <HydraAdmin
        entrypoint={APP_CONSTANTS.ENTRYPOINT}
        dataProvider={dataProvider}
        authProvider={authProvider}
        customRoutes={customRoutes}
        theme={lightTheme}
        lightTheme={lightTheme}
        darkTheme={darkTheme}
        layout={AdminLayout}
        disableTelemetry={true}
        title={'Admin'}
      >
        {
          permissions => typeof permissions !== 'undefined' ? resourceComponents.map((component) => {
            const { resources } = permissions;
            return resources.includes(component.key) ? component : null;
          }) : <React.Fragment />
        }
      </HydraAdmin>
    </Sentry.ErrorBoundary>
  );
};
```
auth provider is not called, so I don't have information about the logged user and permission for this user.

**Comment by 7system7** [comment]
I have the same issue.

**Comment by papppeter** [comment]
Same issue for me.

**Comment by 7system7** [comment]
It seems, the problem is much older... I have a ~6 years old project, and when I delete everything and rebuild the whole FE

```bash
$ pnpm store prune && rm -rf node_modules pnpm-lock.yaml && pnpm i && pnpm dev
```

the auth-provider has never been called anymore. Not w/ the latest, nor w/ the 3.0.0. :disappointed:

If I 
 - restore the original (before this mods) `pnpm-lock.yaml` and reinstall the packages, **the auth-provider works**.
 - I use the `react-admin`'s Admin component, **the auth-provider works**.

**Comment by PawelSuwinski** [comment]
https://github.com/api-platform/admin/issues/550

**Comment by 7system7** [comment]
@PawelSuwinski Thank u! It is working.

**Comment by 7system7** [comment]
Sorry, I was on holiday, and I just quickly checked this solution.

Another problem is persisted. (I do not know anyone has this too.) Somehow the i18provider is not working w/ this solution.

I have a simple polygotProvider almost as the docs said:

```js
import { resolveBrowserLocale } from 'react-admin';
import polyglotI18nProvider from 'ra-i18n-polyglot';
import { enTranslation } from '@yc/core/i18n';

const translations = { en: enTranslation };

export const i18nProvider = polyglotI18nProvider(
  locale => translations?.[locale] ? translations[locale] : translations.en,
  resolveBrowserLocale('en', { fullLocale: true }),
  [
    { locale: 'en', name: 'English' },
  ],
);
```

Translation file is looks like this:

```js
import englishMessages from 'ra-language-english';

export const enTranslation = {
  ...englishMessages,
  resources: {
    users: {
      name: 'User |||| Users',
      fields: {
        partner: 'Partner',

... and so on
```

If I check the first parameter's function w/ a breakpoint in the provider here

```js
  locale => translations?.[locale] ? translations[locale] : translations.en,
                 ^
```

then, I got my translations. But the automatic translations on react-admin's fields w/ or w/o the `label` attribute does not work.

```js
import { TextField } from 'react-admin';

<TextField source="title" label="resources.users.fields.partner" />
```

the `useTranslate` is not working too

```js
import { useTranslate } from 'react-admin';

const t = useTranslate();

console.log(t('resources.users.fields.partner')); // Outputs: resources.users.fields.partner
```

**Comment by PawelSuwinski** [comment]
> Sorry, I was on holiday, and I just quickly checked this solution.
> 
> Another problem is persisted. (I do not know anyone has this too.) Somehow the i18provider is not working w/ this solution.

Yes, because its sanitized and Admin element gets `undefined` that overrides upper context:
https://github.com/api-platform/admin/blob/c04db4848949afe8daeeb36d008f53d3155652f5/src/AdminGuesser.tsx#L110-L117

In this case pass custom ui element, ex.:

```
import {AuthContext, AdminProps, AdminUI } from "react-admin";

const AdminEl = ({authProvider, ...props}: AdminProps) => (
   <AuthContext.Provider value={authProvider}>
     <AdminUI {...props} />
   </AuthContext.Provider>
);

<HydraAdmin 
  admin={AdminEl}
  //...
```

**Comment by 7system7** [comment]
@PawelSuwinski Thank you! Am I thinking correctly that, this is just a workaround and will be fixed in `3.4.7`, without the `admin` property & without the custom element?

**Comment by PawelSuwinski** [comment]
> @PawelSuwinski Thank you! Am I thinking correctly that, this is just a workaround and will be fixed in `3.4.7`, without the `admin` property & without the custom element?

Yes (https://github.com/api-platform/admin/pull/545).
